### PR TITLE
Add benthos-lambda-al2

### PR DESCRIPTION
**Pull request by razor-x**
https://github.com/Jeffail/benthos/issues/895

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -22,6 +22,13 @@ builds:
...
       - CGO_ENABLED=0
     goos: [ linux ]
+    goarch: [ amd64 ]
+  - id: benthos-lambda-al2
+    main: cmd/serverless/benthos-lambda/main.go
+    binary: bootstrap
+    env:
+      - CGO_ENABLED=0
+    goos: [ linux ]
     goarch: [ amd64, arm64 ]
```
Comment (Jeffail) [inline .goreleaser.yml:32]: Does it make sense to include `amd64` here? I don't know anything about al2.

Comment (razor-x) [reply .goreleaser.yml:32]: Yes. To run go on AWS lambda you can choose `go1.x` runtime or `provided.al2` runtime. The `benthos-lambda` build is for `go1.x` which only supports `amd64`, and `benthos-lambda-al2` is for `provided.al2` which supports both `amd64` and `arm64`.

We could have a single build like below that works for  `go1.x` runtime and `provided.al2`, but then we would technically break backwards compatibility by renaming the binary (your call if that's ok):

```yaml
  - id: benthos-lambda
    main: cmd/serverless/benthos-lambda/main.go
    binary: bootstrap
    env:
      - CGO_ENABLED=0
    goos: [ linux ]
    goarch: [ amd64, arm64 ]
  - id: benthos-lambda
    builds: [ benthos-lambda ]
    format: zip
    name_template: "{{ .Binary }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"

Comment (razor-x) [reply .goreleaser.yml:32]: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
image
image

Comment (Jeffail) [reply .goreleaser.yml:32]: Ah okay awesome, thanks for the clarification

**Comment by Jeffail** [comment]
Awesome, thanks @razor-x!

**Pull request merged**